internal/cmd/access/access.go

```diff
@@ -84,28 +90,40 @@ func runAccess(cfg *accessConfig) clierror.Error {
 	return nil
 }
 
-func prepareKubeconfig(cfg *accessConfig) (*api.Config, error) {
-	secret, err := cfg.KubeClient.Static().CoreV1().Secrets(cfg.namespace).Get(cfg.Ctx, cfg.name, metav1.GetOptions{})
+func prepareKubeconfig(cfg *accessConfig) (*api.Config, clierror.Error) {
+	currentCtx := cfg.KubeClient.ApiConfig().CurrentContext
+	clusterName := cfg.KubeClient.ApiConfig().Contexts[currentCtx].Cluster
+	token, duration, err := getServiceAccountToken(cfg)
```

Comment:
shouldn't the token be requested like this only if permanent!=true ?

Reply:
Adressed in followup PR